cli/cmd/init.go

```diff
@@ -0,0 +1,344 @@
+	cmd.Flags().Bool("yes", false, "Accept all defaults")
+	return cmd
+}
+
+func normalizePluginPath(pluginNameOrPath string) (string, error) {
+	parts := strings.Split(pluginNameOrPath, "/")
+	if len(parts) == 1 {
+		return "cloudquery/" + pluginNameOrPath, nil
+	}
+	if len(parts) != 2 {
+		return "", errors.New("invalid plugin path")
+	}
+	return pluginNameOrPath, nil
+}
+
+func parseFlags(cmd *cobra.Command) (source, destination, specPath string, acceptDefaults bool, allErrors error) {
+	source, err := cmd.Flags().GetString("source")
+	allErrors = errors.Join(allErrors, err)
+	if source != "" {
+		source, err = normalizePluginPath(source)
+		allErrors = errors.Join(allErrors, err)
+	}
+	destination, err = cmd.Flags().GetString("destination")
+	allErrors = errors.Join(allErrors, err)
+	if destination != "" {
+		destination, err = normalizePluginPath(destination)
+		allErrors = errors.Join(allErrors, err)
+	}
+	specPath, err = cmd.Flags().GetString("spec-path")
+	allErrors = errors.Join(allErrors, err)
+
+	acceptDefaults, err = cmd.Flags().GetBool("yes")
+	allErrors = errors.Join(allErrors, err)
+	return source, destination, specPath, acceptDefaults, allErrors
+}
+
+func pluginFilter(pluginPath string, kind cqapi.PluginKind) func(plugin cqapi.ListPlugin) bool {
+	return func(plugin cqapi.ListPlugin) bool {
+		return plugin.TeamName+"/"+plugin.Name == pluginPath && plugin.Kind == kind && plugin.LatestVersion != nil
+	}
+}
+
+func pluginName(plugin cqapi.ListPlugin, _ int) string {
+	return plugin.Name
+}
+
+func officialReleasedPluginsByKind(kind cqapi.PluginKind) func(plugin cqapi.ListPlugin, _ int) bool {
+	return func(plugin cqapi.ListPlugin, _ int) bool {
+		return plugin.Kind == kind && plugin.Official && plugin.ReleaseStage != cqapi.PluginReleaseStageComingSoon && plugin.LatestVersion != nil
+	}
+}
+
+func pluginsSorter(plugins []cqapi.ListPlugin, prioritySlice []string) func(a, b int) bool {
+	return func(a, b int) bool {
+		indexOfA := lo.IndexOf(prioritySlice, plugins[a].Name)
+		indexOfB := lo.IndexOf(prioritySlice, plugins[b].Name)
+		if indexOfA == -1 && indexOfB != -1 {
+			return false
+		}
+		if indexOfA != -1 && indexOfB == -1 {
+			return true
+		}
+		if indexOfA == -1 && indexOfB == -1 {
+			return plugins[a].Name < plugins[b].Name
+		}
+		return indexOfA < indexOfB
+	}
+}
+
+func extractYamlFromMarkdownCodeBlock(markdown string) string {
+	re := regexp.MustCompile("```yaml.*?\n([\\s\\S]+?)\n```")
+
+	matches := re.FindStringSubmatch(markdown)
+	if len(matches) < 2 {
+		return ""
+	}
+
+	return matches[1]
+}
+
+func defaultConfigForPlugin(plugin cqapi.ListPlugin) *strings.Builder {
+	sb := strings.Builder{}
+	sb.WriteString("kind: " + string(plugin.Kind) + "\n")
+	sb.WriteString("spec:\n")
+	sb.WriteString("  name: " + plugin.Name + "\n")
+	sb.WriteString("  path: " + plugin.TeamName + "/" + plugin.Name + "\n")
+	sb.WriteString("  version: " + *plugin.LatestVersion + "\n")
+	return &sb
+}
```

Comment:
I think it's more maintainable to make this more readable, but I leave that to your discretion, I don't know how much we expect this to change.

I'd have two separate functions for source & destination (so that you don't have the source spec in two different places, and they can evolve separately in the future at the cost of a little repetition now), and use text/template instead of a string builder.

If you want to make the change, here's the rewrite for this one:

```go
func defaultConfigForPlugin(plugin cqapi.ListPlugin) *strings.Builder {
	tmpl := `kind: {{.Kind}}
spec:
  name: {{.Name}}
  path: {{.TeamName}}/{{.Name}}
  version: {{.LatestVersion}}
`
	var buf bytes.Buffer
	t := template.Must(template.New("config").Parse(tmpl))
	t.Execute(&buf, plugin)

	sb := strings.Builder{}
	sb.WriteString(buf.String())
	return &sb
}
```

Reply:
Good idea, refactored in https://github.com/cloudquery/cloudquery/pull/19016/commits/ac53790416a338fd30fa5ebc0b75766d6267650f and added tests